ldpd/lde.c

```diff
@@ -1708,6 +1727,34 @@ on_get_label_chunk_response(uint32_t start, uint32_t end)
 		current_label_chunk = listtail(label_chunk_list);
 }
 
+void
+lde_free_label(uint32_t label)
+{
+	struct listnode     *node;
+	struct label_chunk  *label_chunk;
+	uint64_t		     pos;
+
+	node = listhead(label_chunk_list);
+
+	while (node) {
+		label_chunk = listgetdata(node);
+
+		if (label_chunk) {
+			if ((label < label_chunk->end)  && (label >= label_chunk->start)) {
+				pos = 1 << (label - label_chunk->start);
+				label_chunk->used_mask &= ~pos;
+				/*nobdy using this chunk..free it if its not crrent_label_chunk*/
+				if (!label_chunk->used_mask && (current_label_chunk != node))
```

Comment:
As above, it'd be clearer to me if the management of the global list and the chunk's allocated memory were dealt with right here.

Reply:
OK.. I will change it like below..

               /*nobody using this chunk..free it if it's not current_label_chunk*/
                if (!label_chunk->used_mask && (current_label_chunk != node))
                {
                    if (lde_release_label_chunk(label_chunk->start, label_chunk->end) != 0)
                        log_warn("%s: Error releasing label chunk!", __func__);
                    else
                    {
                        listnode_delete(label_chunk_list, node);
                        lde_del_label_chunk(label_chunk);
                    }
                }